src/ExpressionEvaluator/Core/Test/ResultProvider/Debugger/Engine/DkmClrType.cs

```diff
@@ -20,6 +20,16 @@ namespace Microsoft.VisualStudio.Debugger.Clr
     [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
     public class DkmClrType
     {
+        /// <summary>
+        /// We would accept inherited members for tests purposes comparing to <see cref="TypeHelpers.MemberBindingFlags"/> 
+        /// because an actual VS <see cref="GetEvalAttributes(Type)"/> may return attributes from base types.
+        /// Therefore, we do not check here for <see cref="BindingFlags.DeclaredOnly"/>.
+        /// </summary>
+        internal const BindingFlags MemberBindingFlags = BindingFlags.Public |
+                                                         BindingFlags.NonPublic |
+                                                         BindingFlags.Instance |
+                                                         BindingFlags.Static;
```

Comment:
Does `DkmClrType.GetEvalAttributes()` return the attributes for members of the base types? The methods in `TypeHelpers` that rely on `DkmClrType.GetEvalAttributes()` walk up the type hierarchy calling `GetEvalAttributes()` on each type until an appropriate attribute is found, but if the first call (with the most derived type) returned all attributes, the subsequent calls are unnecessary.

Reply:
I also reviewed code for other attributes. It seems we process the properly: accepting multiple attributes provided by the Debugger code.

---
In reply to: [251983321](https://github.com/dotnet/roslyn/pull/26049#discussion_r251983321) [](ancestors = 251983321,251189134)